pkg/homedir/homedir_unix.go

```diff
@@ -101,9 +101,39 @@ func isWriteableOnlyByOwner(perm os.FileMode) bool {
 	return (perm & 0o722) == 0o700
 }
 
-// GetConfigHome returns XDG_CONFIG_HOME.
+// ConfigHome returns XDG_CONFIG_HOME.
+// ConfigHome returns $HOME/.config and nil error if XDG_CONFIG_HOME is not set.
+// Verifies ownership of config directory matches the current process or
+// returns error.
+// See also https://standards.freedesktop.org/basedir-spec/latest/ar01s03.html
+func GetConfigHome() (string, error) {
+	rootlessConfigHomeDirOnce.Do(func() {
+		tmpDir := os.Getenv("XDG_CONFIG_HOME")
+		if tmpDir == "" {
+			home := Get()
+			resolvedHome, err := filepath.EvalSymlinks(home)
+			if err != nil {
+				rootlessConfigHomeDirError = fmt.Errorf("cannot resolve %s: %w", home, err)
+				return
+			}
+			tmpDir = filepath.Join(resolvedHome, ".config")
+		}
+		st, err := os.Stat(tmpDir)
+		if err == nil {
```

Comment:
If the dir is missing, this implementation sets (…HomeDir, HomeDirError) to (`""`, `nil`), which seems undesirable to me.

Reply:
Still outstanding.

Reply:
My mistake, this now propagates `ENOENT` to the caller. That’s discussed elsewhere…